### PR TITLE
Add interfaces and initial tests for Multi-device conversation

### PR DESCRIPTION
**Pull request by jk-ms**
Start adding in support for the Multi-device Conversation feature:

### CODE REVIEW

File: src/sdk/Exports.ts [hunk 1]
```diff
@@ -50,3 +50,17 @@ export { ActivityReceivedEventArgs } from "./ActivityReceivedEventArgs";
 export { ServicePropertyChannel } from "./ServicePropertyChannel";
 export { ProfanityOption } from "./ProfanityOption";
 export { BaseAudioPlayer } from "./Audio/BaseAudioPlayer";
+export { Conversation,
+    ConversationExpirationEventArgs,
+    ConversationImpl,
```
Comment (rhurey) [inline src/sdk/Exports.ts]: >ConversationImpl [](start = 4, length = 16)

Why are we exporting a Impl class on the public SDK interface?

File: tests/ConversationTranslatorTests.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    ConsoleLoggingListener
+} from "../src/common.browser/Exports";
+import {
+    Events,
+    EventType,
+} from "../src/common/Exports";
+import { Settings } from "./Settings";
+
+// tslint:disable-next-line:no-console
+const consoleInfo = console.info;
```
Comment (rhurey) [inline tests/ConversationTranslatorTests.ts]: Nice. I need to remember this....

File: src/sdk/Exports.ts (hunk 1 above)
Comment (jk-ms) [reply src/sdk/Exports.ts]: Updated code to remove this export as it's not required here.

File: src/sdk/Exports.ts [hunk 3]
```diff
@@ -50,3 +50,17 @@ export { ActivityReceivedEventArgs } from "./ActivityReceivedEventArgs";
...
+    ConversationParticipantsChangedEventArgs,
+    ConversationTranslationCanceledEventArgs,
+    ConversationTranslationEventArgs,
+    ConversationTranslationResult,
+    ConversationTranslator,
+    IConversation,
```
Comment (rhurey) [inline src/sdk/Exports.ts]: >IConversation [](start = 4, length = 13)

Why expose both the interface and the abstract class?

Comment (jk-ms) [reply src/sdk/Exports.ts]: Do you mean for the Conversation / IConversation?
If I've got a Typescript client consuming the sdk, then I get a no exported member error if they aren't both exported, e.g.
const myConversation: sdk.IConversation = sdk.Conversation.createConversationAsync(myConfig);

Comment (jk-ms) [reply src/sdk/Exports.ts]: The interface is removed from the exports.ts

File: src/sdk/ConversationTranslator/Conversation.ts [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+import { PropertyCollection, SpeechTranslationConfig } from "../Exports";
+import { IConversation } from "./IConversation";
+import { IParticipant, IUser } from "./IParticipant";
+
+export abstract class Conversation implements IConversation {
+
+    public abstract get authorizationToken(): string;
+    public abstract set authorizationToken(value: string);
+
+    public abstract get config(): SpeechTranslationConfig;
```
Comment (ralph-msft) [inline src/sdk/ConversationTranslator/Conversation.ts]: Please remove these. Setting the SpeechConfig after a conversation has started is not a supported operation. Getting the SpeechConfig and then changing a property there also leads to an expectation that would change the Conversation which it does not.

File: src/sdk/ConversationTranslator/Conversation.ts [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+    public abstract set config(value: SpeechTranslationConfig);
```
Comment (ralph-msft) [inline src/sdk/ConversationTranslator/Conversation.ts]: Please remove (see above)

**Pull request merged**